scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/DataSpaceNodeClient.java

```diff
@@ -0,0 +1,172 @@
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ *  * $$ACTIVEEON_INITIAL_DEV$$
+ */
+package org.ow2.proactive.scheduler.task.client;
+
+
+import org.objectweb.proactive.annotation.PublicAPI;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.ow2.proactive.scheduler.common.exception.PermissionException;
+import org.ow2.proactive.scheduler.common.task.dataspaces.FileSystemException;
+import org.ow2.proactive.scheduler.common.task.dataspaces.RemoteSpace;
+import org.ow2.proactive.scheduler.rest.ds.DataSpaceClient;
+import org.ow2.proactive.scheduler.rest.ds.IDataSpaceClient;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * DataSpace api available as a variable during the execution of a task. it is based on the RemoteSpace interface.
+ *
+ * @author ActiveEon Team
+ */
+@PublicAPI
+public class DataSpaceNodeClient implements RemoteSpace {
+
+    SchedulerNodeClient schedulerNodeClient;
+    DataSpaceClient dataSpaceClient;
+    IDataSpaceClient.Dataspace space;
+    String schedulerRestUrl;
+
+    RemoteSpace spaceProxy;
+
+
+    public DataSpaceNodeClient(SchedulerNodeClient schedulerNodeClient, IDataSpaceClient.Dataspace space, String schedulerRestUrl) {
+        this.schedulerNodeClient = schedulerNodeClient;
+        this.dataSpaceClient = new DataSpaceClient();
+        this.space = space;
+        this.schedulerRestUrl = schedulerRestUrl;
+        switch (space) {
+            case GLOBAL:
+                spaceProxy = dataSpaceClient.getGlobalSpace();
+                break;
+            case USER:
+                spaceProxy = dataSpaceClient.getUserSpace();
+                break;
+            default:
+                throw new IllegalStateException("Unkown space : " + space);
+        }
+    }
+
+    /**
+     * Connects to the dataspace at the default schedulerRestUrl, using the current user credentials
+     *
+     * @throws Exception
+     */
+    public void connect() throws Exception {
+        connect(schedulerRestUrl);
+    }
+
+    /**
+     * Connects to the dataspace at the specified schedulerRestUrl, using the current user credentials
+     *
+     * @param url schedulerRestUrl of the scheduler
+     * @throws Exception
+     */
+    public void connect(String url) throws Exception {
+        schedulerNodeClient.connect(url);
+        this.dataSpaceClient.init(url, schedulerNodeClient);
+    }
+
+    private void renewSession() {
+        try {
+            schedulerNodeClient.renewSession();
+        } catch (NotConnectedException e) {
+            throw new RuntimeException(e);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)